src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -279,4 +290,86 @@ private SafeSslHandle() : base(IntPtr.Zero, true)
         {
         }
     }
+
+    internal sealed class SafeChannelBinding : SafeHandle
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        private struct SEC_CHANNEL_BINDINGS
+        {
+            internal int dwInitiatorAddrType;
+            internal int cbInitiatorLength;
+            internal int dwInitiatorOffset;
+            internal int dwAcceptorAddrType;
+            internal int cbAcceptorLength;
+            internal int dwAcceptorOffset;
+            internal int cbApplicationDataLength;
+            internal int dwApplicationDataOffset;
+        }
+
+        private static readonly byte[] s_tlsServerEndPointByteArray = Encoding.UTF8.GetBytes("tls-server-end-point:");
+        private static readonly byte[] s_tlsUniqueByteArray = Encoding.UTF8.GetBytes("tls-unique:");
+        private static readonly int secChannelBindingSize = Marshal.SizeOf<SEC_CHANNEL_BINDINGS>();
+        private readonly int cbtPrefixByteArraySize;
+        private const int CertHashMaxSize = 128;
+
+        internal int Length
+        {
+            get;
+            private set;
+        }
+
+        internal IntPtr CertHashPtr
+        {
+            get;
+            private set;
+        }
+
+        private byte[] GetPrefixBytes(ChannelBindingKind kind)
+        {
+            if (kind == ChannelBindingKind.Endpoint)
+            {
+                return s_tlsServerEndPointByteArray;
+            }
+            else
```

Comment:
There are three ChannelBindingKinds.  Your else can assert that the kind is Unique; but otherwise this code is neither resilient to changes in the future, nor resilient to an Unknown slipping in.